### ISSUE TITLE
NullReferenceException when accessing lazy properties of HttpRequest (only) with AI enabled

### ISSUE THREAD

**Issue by silmon27** [microsoft/ApplicationInsights-dotnet#1635]
##  Behavior
I'm getting the following exception every couple of minutes/hours where we access HttpRequest.
```
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.AspNetCore.Http.DefaultHttpRequest.get_Query()
```
If I recall correctly this started when we upgraded to ASP Core 3.0 (may be coincidence since we also updated AI and some other things at that time).
It happens in multiple microservices of ours (all running aspnet:3.1.0-alpine). Some very simple, some more complex. We don't use `IHttpContextAccessor` and I unless I missed something we aren't using the `HttpContext` concurrently.
As soon as I remove `services.AddApplicationInsightsTelemetry()` the exceptions stop.

## Repro Steps
Unfortunately I'm not able to share our project code to repro.
This happens only/more often in services with high traffic/lots of concurrent requests.

The only other thing I can think of is that we are also using Serilog for logging (latest version), but only Warning and above. I don't really see how that could possibly interfere with AI.

## Version Info 
SDK Version: Microsoft.ApplicationInsights 2.12.0 & Microsoft.ApplicationInsights.AspNetCore 2.12.0
.NET Version: ASP.NET Core 3.1
How Application was onboarded with SDK: NuGet Package
OS: Linux aspnet:3.1.0-alpine container
Hosting Info: Azure Kubernetes

**Comment by Dmitry-Matveev** [comment]
@silmon27 , thanks for reporting! Can you please share the full stack of the exception or at least a part with Application Insights methods?

**Comment by silmon27** [comment]
These exceptions are thrown in our code where we access HttpRequest (query, form, ...). There is no Application Insights in the stack, just our own code.
Even directly in
```
public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
{
  // app.UseExceptionPage, app.UseExceptionLogging, app.UseResponseCompression, etc.

  app.Run(async context => {
    // NullReferenceException might get thrown when accessing HttpRequest here (e.g. context.Request.Query["key"])
  });
}
```
with very little middleware.

Since we disabled AI a month ago we haven't seen this issue occur anymore, so it must definitely be related somehow.
Any ideas?

**Comment by Dmitry-Matveev** [comment]
@silmon27 , I'm transitioning this issue, new owner will pick it up.
Current state is:
- Microsoft.AspNetCore.Http.DefaultHttpRequest.get_Query sometimes throws an exception in user stack with no AI methods on the stack, however, as soon as AI is removed, exceptions are no longer thrown;
- I reviewed usages of Request object in our code to check if we may dispose of it accidentally or replace with `null` on HTTP Context, or we have some extension method on HttpContext that may have negative consequences, such as changing Request object - so far nothing. Another track I was pursuing, but haven't closed on that yet - review `get_Query()` implementation and check whether we affect some object inside the chain, e.g. we use `get_Query()` ourselves and we have methods that operate on Request object, so some internal state may still be changed.

**Comment by rajkumar-rangaraj** [comment]
@silmon27 I continued investigation from where it was left off. I did not find any issue and could not reproduce it too. 

It would be very difficult to identify this without a repro. Does on-boarding the application insights to same app bring back the same issue?

**Comment by silmon27** [comment]
@rajkumar-rangaraj I tried again today with updated libraries (2.13.1). I simply uncommented `services.AddApplicationInsightsTelemetry()` and a couple of minutes later the same exceptions started flowing in again (roughly every 10 minutes).

I understand that it is very difficult to identify this without repro code. If I get the time, I'll try to create a repro. But that might take a while since I can't even consistently trigger it myself in production. Seems like there is some magical combination of X (concurrent) requests taking X amount of time or being X ms close to each other. The URL doesn't seem to matter, happens everywhere where we access `get_Query` (or `get_Form`).
